### PR TITLE
Refactor to workflow tb api

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Added

-

### Changed

- Refactor to workflow tb api

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b refactor-to-workflow-tb-api -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by ChrOertlin — APPROVED** [review]
💯

File: cg/meta/workflow/analysis.py [hunk 1]
```diff
@@ -209,14 +209,14 @@ def add_pending_trailblazer_analysis(self, case_id: str) -> None:
...
             analysis_type=self.get_application_type(
                 self.status_db.get_case_by_internal_id(case_id).links[0].sample
             ),
-            out_dir=self.get_job_ids_path(case_id).parent.as_posix(),
             config_path=self.get_job_ids_path(case_id).as_posix(),
+            out_dir=self.get_job_ids_path(case_id).parent.as_posix(),
             slurm_quality_of_service=self.get_slurm_qos_for_case(case_id),
-            data_analysis=str(self.workflow),
+            email=environ_email(),
+            workflow=self.workflow,
```
Comment (islean) [inline cg/meta/workflow/analysis.py:219]: Will this be serialized in a json? Thinking if we might have a serialization issue with the subtle difference between str(enum) and enum.

File: cg/apps/tb/api.py [hunk 2]
```diff
@@ -120,13 +120,14 @@ def add_pending_analysis(
             "config_path": config_path,
             "out_dir": out_dir,
             "priority": slurm_quality_of_service,
-            "data_analysis": str(data_analysis).upper(),
+            "workflow": workflow.upper(),
```
Comment (islean) [inline cg/apps/tb/api.py:123]: Same question here

File: cg/meta/workflow/analysis.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/meta/workflow/analysis.py:219]: Seem to work like this in testing. Let's deploy but be aware of this if issues arise.

**Comment by henrikstranneheim** [comment]
Deployed in production:
```
Log deploy... done.
cg, version 59.0.0

**Pull request merged**